src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -469,8 +469,48 @@ public bool TryGetDateTime(out DateTime value)
                 throw ThrowHelper.GetInvalidOperationException_ExpectedString(TokenType);
             }
 
-            ReadOnlySpan<byte> span = HasValueSequence ? ValueSequence.ToArray() : ValueSpan;
-            return JsonHelpers.TryParseAsISO(span, out value, out int bytesConsumed) && span.Length == bytesConsumed;
+            ReadOnlySpan<byte> span = ValueSpan;
+
+            if (HasValueSequence)
+            {
+                int sequenceLength = (int)ValueSequence.Length;
+
+                if (!JsonReaderHelper.IsValidDateTimeParsingLength(sequenceLength))
+                {
+                    value = default;
+                    return false;
+                }
+
+                Debug.Assert(sequenceLength <= JsonConstants.MaximumEscapedDateTimeOffsetParseLength);
+                Span<byte> stackSpan;
+                // Cannot create a span directly since it gets passed to instance methods on a ref struct.
```

Comment:
I don't think there are any pass-to-ref-struct cases in this routine (including calls on `this`), so you can simplify things.

```C#
Span<byte> stackSpan = stackalloc byte[sequenceLength];
ValueSequence.CopyTo(stackSpan);
```

You might need to change how `span` is declared to be

```C#
ReadOnlySpan<byte> span = stackalloc byte[0];

if (HasValueSequence)
{
    ...
}
else
{
    span = ValueSpan;
}
```

So that `span` gets the "I'm allowed to hold stackalloc" marker.

Reply:
Nice. Thanks.